ui-components/src/page_components/listing/ListingCard.tsx

```diff
@@ -22,10 +25,75 @@ export interface ListingCardProps {
   tableHeaderProps?: ListingCardHeaderProps
   tableProps: ListingCardTableProps
   detailsLinkClass?: string
+  listing: Listing
 }
 
 const ListingCard = (props: ListingCardProps) => {
   const { imageCardProps, tableProps, detailsLinkClass, tableHeaderProps } = props
+  const { profile, userProfileService } = useContext(AuthContext)
+  const [updatedFavorites, setUpdatedFavorites] = useState(profile?.preferences?.favoriteIDs)
+
+  useEffect(() => {
+    setFavoriteButtonState()
+  })
+
+  const setFavoriteButtonState = () => {
+    const preferences: UserPreferences = profile?.preferences || { favoriteIDs: [] }
+    setUpdatedFavorites(preferences?.favoriteIDs)
+  }
+
+  const addToFavorite = async () => {
+    if (!profile) {
+      return
+    }
+    const preferences: UserPreferences = profile?.preferences || { favoriteIDs: [] }
+    if (!preferences.favoriteIDs) {
+      preferences.favoriteIDs = []
+    }
+
+    if (!preferences.favoriteIDs.includes(props.listing.id)) {
```

Comment:
Might be a little more clear to invert the `if` statement here and early return:

```
if (preferences.favoriteIDs.includes(props.listing.id)) {
  return
}

preferences.favoriteIDs.push(props.listing.id)
```

Reply:
I like this suggestion.  Easier to understand what's happening.  Thanks. Updated.